src/OrbitBase/LoggingUtils.cpp

```diff
@@ -11,68 +11,96 @@
 #include <system_error>
 
 #include "OrbitBase/Logging.h"
+#include "OrbitBase/SafeStrerror.h"
+
+namespace orbit_base_internal {
 
 const absl::Duration kLogFileLifetime = absl::Hours(24 * 7);  // one week
 // Determined by kLogFileNameDelimiter, i.e., the format of log file names.
 const int kTimestampStartPos = 6;
 // Determined by kLogFileNameTimeFormat, i.e., the format of timestamp contained in log file names.
 const int kTimestampStringLength = 19;
 
-std::vector<std::filesystem::path> ListFiles(const std::filesystem::path& log_dir) {
+std::vector<std::filesystem::path> ListFilesRecursivelyIgnoreErrors(
+    const std::filesystem::path& dir) noexcept {
   std::vector<std::filesystem::path> files_in_dir;
-  for (const auto& file_path : std::filesystem::recursive_directory_iterator(log_dir)) {
-    if (file_path.is_regular_file()) {
-      files_in_dir.push_back(file_path.path());
+  std::error_code error;
+  auto directory_iterator = std::filesystem::recursive_directory_iterator(dir, error);
+  if (error) {
+    ERROR("Unable to open directory \"%s\": %s", dir.string(), error.message());
+    return {};
+  }
+
+  for (auto it = std::filesystem::begin(directory_iterator),
+            end = std::filesystem::end(directory_iterator);
+       it != end; it.increment(error)) {
+    if (error) {
+      ERROR("directory_iterator::increment failed for \"%s\": %s (will ignore)", dir.string(),
+            error.message());
+      continue;
+    }
+    bool is_regular_file = it->is_regular_file(error);
+    if (error) {
+      ERROR("Unable to stat \"%s\": %s (will ignore)", it->path().string(), error.message());
+      continue;
+    }
+
+    if (is_regular_file) {
+      files_in_dir.push_back(it->path());
     }
   }
+
   return files_in_dir;
 }
 
-ErrorMessageOr<absl::Duration> GetLogFileLifetime(const std::string& log_file_name) {
+ErrorMessageOr<absl::Time> ParseLogFileTimestamp(const std::string& log_file_name) noexcept {
   if (log_file_name.size() < kTimestampStartPos + kTimestampStringLength) {
     return ErrorMessage(
         absl::StrFormat("Unable to extract time information from log file: %s", log_file_name));
   }
   std::string timestamp_string = log_file_name.substr(kTimestampStartPos, kTimestampStringLength);
   absl::Time log_file_timestamp;
   std::string parse_time_error;
-  if (!absl::ParseTime(kLogFileNameTimeFormat, timestamp_string, absl::LocalTimeZone(),
+  if (!absl::ParseTime(kLogFileNameTimeFormat, timestamp_string, absl::UTCTimeZone(),
```

Comment:
I am unsure about UTCTimeZone. I would agree if we used a iso timestamp like `2021-02-16T16:24:47Z`. The Z indicates about UTC. Maybe we should change the format, or stick to local time.. wdyt?

Reply:
I went with UTC because it does not require knowledge of local timezones, daylight saving times and it is universally transferable. You can reason about the time the file was created without asking the person who sent it "what is your timezone?"

Reply:
okay. I guess its better, even without the Z